aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -58,199 +59,206 @@ namespace {
 
 } // end anonymous namespace
 
-void _validate_sparse_compressed_tensor_args_worker(const Tensor& compressed_indices, const Tensor& plain_indices, const Tensor& values, const IntArrayRef size, const Layout& layout) {
+/*
+  Validate the arguments to sparse compressed (CSR, CSC, BSR, and BSC)
+  tensor factory functions.
+
+  The CSR and BSR invariants for PyTorch are outlined in
+
+    https://pearu.github.io/csr_tensor_invariants.html
```

Comment:
It mentions that `int32` is default for CSR, which is not the case for COO. I have also not seen any indexing tensors which are not `int64` in the dense case.

Reply:
https://pearu.github.io/csr_tensor_invariants.html is a little out-of-date: `int64` is the current default for CSR tensors.

The `int32` index support originates from `MKL_INT` being 32-bit integer, see [comment](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/SparseCsrTensorImpl.h#L21-L23).